src/modalities/nn/weight_init/high_level_weight_init_factory.py

```diff
@@ -0,0 +1,132 @@
+from typing import List, Optional
+
+from pydantic import BaseModel, Field, model_validator
+from typing_extensions import Annotated
+
+from modalities.config.pydanctic_if_types import PydanticWeightInitializationIFType
+from modalities.nn.weight_init.low_level_weight_init_factory import LowLevelInitializationFactory
+from modalities.nn.weight_init.parameter_name_regex_filters import (
+    NAMED_PARAMETER_INIT_GROUPS,
+    SupportWeightInitModels,
+    WeightInitTypes,
+)
+from modalities.nn.weight_init.weight_init import WeightInitializationIF, WeightInitializerWrapper
+
+
+class WeightInitializerWrapperConfig(BaseModel):
+    weight_initializers: List[PydanticWeightInitializationIFType]
+
+
+class ComposedWeightInitializationConfig(BaseModel):
+    model_type: SupportWeightInitModels
+    weight_init_type: WeightInitTypes
+
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    hidden_dim: Optional[Annotated[int, Field(strict=True, gt=0)]] = None
+    num_layers: Optional[Annotated[int, Field(strict=True, gt=0)]] = None
+
+    @model_validator(mode="after")
+    def _check_values(self):
+        # in case of initialization with "auto", we need to specify the hidden_dim
+        if self.std == "auto" and self.hidden_dim is None:
+            raise ValueError("hidden_dim must be specified when std is 'auto'")
```

Comment:
```suggestion
            raise ValueError("hidden_dim must be specified when std is 'auto'")
            
        if self.std != "auto" and self.hidden_dim is not None:
            raise ValueError("hidden_dim must not be specified when std is a number")
```

Reply:
added pydantic checks for this case:
https://github.com/Modalities/modalities/commit/eea99bc1ba86923c8e9fbfa410c8bc502e8598f4